cocotb_test/simulator.py

```diff
@@ -94,16 +94,30 @@ def __init__(
         self.toplevel = toplevel
         self.toplevel_lang = toplevel_lang
 
-        if verilog_sources is None:
-            verilog_sources = []
-
+        if verilog_sources:
+            if isinstance(verilog_sources, list):
+                # create named library with toplevel module as its name
+                verilog_sources = {f"{self.toplevel_module}": verilog_sources}
+                # format toplevel as '<lib>.<module>'
+                if not "." in self.toplevel:
+                    self.toplevel = ".".join((self.toplevel, self.toplevel))
+        else:
+            verilog_sources = dict()  # empty dict instead of empty list
```

Comment:
This can be a function? Same as below?

Reply:
Like this?

Reply:
Was rather thinking to make a function of repeating part but is ok.

```python
            if isinstance(self.vhdl_sources, list):
                # create named library with toplevel module as its name
                self.vhdl_sources = {f"{self.toplevel_module}": self.vhdl_sources}
                # format toplevel as '<lib>.<module>'
                if not "." in self.toplevel:
                    self.toplevel = ".".join((self.toplevel, self.toplevel))
```